components/main-chef-wrapper/main.go

```diff
@@ -0,0 +1,123 @@
+//
+// Copyright 2019 Chef Software, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+)
+
+func main() {
+	// No arguments provided, display usage
+	if len(os.Args) <= 1 {
+		usage()
+		os.Exit(0)
+	}
+	var (
+		subCommand = os.Args[1]
+		allArgs    = os.Args[2:]
```

Comment:
It might be worth it to spike an an early implementation of `-v` and `-h` just so you don't get too far down the road without figuring out how to handle partial option processing.  In some other wrappers, I've often resorted to just doing manual handling of the handful of top-level options since most libraries aren't good at parsing only a subset of options.

Reply:
@stevendanna do you mean to catch when the user provides either `chef -v` or `chef -h` commands? 🤔

Reply:
I see your +1 @stevendanna - It makes sense to spike on that implementation, especially since this top-level command will allow us to standardize a lot of those flags and or config details, though, we would like to be as transparent as possible in this first iteration to the end-user, have feedback about the new integration with the new tools we will create and see if it makes sense to start implementing and improving existing functionality. 

Do you see any other blocker for us to merge and continue iterating on this approach/idea?